chromadb/api/__init__.py

```diff
@@ -167,6 +167,30 @@ def _update(
         """
         pass
 
+    @abstractmethod
+    def _upsert(
+        self,
+        collection_name: str,
+        ids: IDs,
+        embeddings: Optional[Embeddings] = None,
+        metadatas: Optional[Metadatas] = None,
+        documents: Optional[Documents] = None,
+        increment_index: bool = True,
+    ):
+        """Add or update entries in the embedding store. 
+        If an entry with the same id already exists, it will be updated, otherwise it will be added.
+        ⚠️ This operation is slower than add because it needs to check if the entry already exists.
```

Comment:
Nit: This probably shouldn't live at the base class level since its an implementation level comment.